### PR TITLE
ci: fix e2e jobs for `Istio 1.11`

### PR DESCRIPTION
**Pull request by yskopets**
Signed-off-by: Yaroslav Skopets <yaroslav@tetrate.io>

## Summary

* this PR fixes e2e jobs for `Istio 1.11`
* to be clear, it doesn't mean that all e2e jobs will always be successful
  * some tests are fine on `GKE 1.19` but fail on more recent `GKE` versions
  * there is an element of flakiness, anyway
* this PR aims to achieve that there will be at least 1 successful job on `GKE` and 1 on `EKS`

## Examples

* CI on [tetrate-release-1.11](https://github.com/tetratelabs/istio/runs/5145932880?check_suite_focus=true)
* CI on [tetratefips-release-1.11](https://github.com/tetratelabs/istio/actions/runs/1825470220)

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -210,6 +210,7 @@ jobs:
         env:
           K8S_VERSION: ${{ matrix.version }}
           ISTIO_MINOR_VER: ${{ steps.get_minor_ver.outputs.REL_BRANCH_VER }}
+          EKS_AWAILABILITY_ZONES: ${{ secrets.EKS_AWAILABILITY_ZONES }}
```
Comment (chirauki) [inline .github/workflows/e2e_tests.yml]:
```suggestion
          EKS_AVAILABILITY_ZONES: ${{ secrets.EKS_AVAILABILITY_ZONES }}
```

File: tetrateci/create_eks_cluster.sh [hunk 2]
```diff
@@ -27,5 +27,19 @@ SHA8=$(git rev-parse --short $GITHUB_SHA)
...
+# sufficient capacity to support the cluster.
+# Retry and choose from these availability zones: us-east-1a, us-east-1b, us-east-1c, us-east-1d, us-east-1f
+# ```
+#
+# As a workaround, be ready to set availability zones explicitly.
+EKS_AWAILABILITY_ZONES=${EKS_AWAILABILITY_ZONES:-}
+
 echo "creating a eks cluster with \"$CLUSTER_NAME\" name..."
-eksctl create cluster --name $CLUSTER_NAME --version $K8S_VERSION --nodes 3 --node-type m5.xlarge
+eksctl create cluster --name $CLUSTER_NAME --version $K8S_VERSION --nodes 3 --node-type m5.xlarge "--zones=${EKS_AWAILABILITY_ZONES}"
```
Comment (chirauki) [inline tetrateci/create_eks_cluster.sh]:
```suggestion
eksctl create cluster --name $CLUSTER_NAME --version $K8S_VERSION --nodes 3 --node-type m5.xlarge "--zones=${EKS_AVAILABILITY_ZONES}"
```

File: tetrateci/patches/eks/eks-ingress.1.11.patch [hunk 3]
```diff
@@ -59,69 +59,3 @@ index 2453b16fd3..2cbe717afc 100644
  					if got != host {
  						return fmt.Errorf("unexpected ingress status, got %+v want %v", got, host)
  					}
-diff --git a/tetrateci/patches/eks/eks-ingress.1.11.patch b/tetrateci/patches/eks/eks-ingress.1.11.patch
-index 5585df7a68..e69de29bb2 100644
---- a/tetrateci/patches/eks/eks-ingress.1.11.patch
-+++ b/tetrateci/patches/eks/eks-ingress.1.11.patch
```
Comment (chirauki) [inline tetrateci/patches/eks/eks-ingress.1.11.patch:65]: is it again a patch that patches itself?

Comment (yskopets) [reply tetrateci/patches/eks/eks-ingress.1.11.patch:65]: yeah, this PR removes the part that patches itself

File: tetrateci/create_eks_cluster.sh [hunk 4]
```diff
@@ -27,5 +27,19 @@ SHA8=$(git rev-parse --short $GITHUB_SHA)
...
+# sufficient capacity to support the cluster.
+# Retry and choose from these availability zones: us-east-1a, us-east-1b, us-east-1c, us-east-1d, us-east-1f
+# ```
+#
+# As a workaround, be ready to set availability zones explicitly.
+EKS_AWAILABILITY_ZONES=${EKS_AWAILABILITY_ZONES:-}
+
 echo "creating a eks cluster with \"$CLUSTER_NAME\" name..."
-eksctl create cluster --name $CLUSTER_NAME --version $K8S_VERSION --nodes 3 --node-type m5.xlarge
+eksctl create cluster --name $CLUSTER_NAME --version $K8S_VERSION --nodes 3 --node-type m5.xlarge "--zones=${EKS_AVAILABILITY_ZONES}"
```
Comment (yskopets) [inline tetrateci/create_eks_cluster.sh]: @chirauki BTW, CloudFormation started experiencing permanent problems with deleting a VPC after I made 2 changes:
* changed `AWS_REGION` from `ap-south-1` to `us-east-1` (`ap-south-1` was causing unacceptable latency in tests)
* set `EKS_AVAILABILITY_ZONES` to `us-east-1a,us-east-1b` (as a workaround for `eksctl` unable to choose proper availability zones automatically)

So, I'm now trying to set `AWS_REGION=us-east-2` and unset `EKS_AVAILABILITY_ZONES`

Comment (chirauki) [reply tetrateci/create_eks_cluster.sh]: yes, let's avoid `us-east-1`. Historically, is the region with more problems.

Comment (yskopets) [reply tetrateci/create_eks_cluster.sh]: After moving to `us-east-2`, there is no more problems with VPC deletion.

And I'm no longer using `EKS_AVAILABILITY_ZONES`.

File: tetrateci/patches/eks/eks-ingress.1.11.patch (hunk 3 above)
Comment (chirauki) [reply tetrateci/patches/eks/eks-ingress.1.11.patch:65]: oh... sh... I missread this one.

**Review by deva26 — APPROVED** [review]
Thanks.

**Pull request merged**